api/types/apicache.go

```diff
@@ -476,7 +476,7 @@ func (pq *BlockPriorityQueue) Insert(summary *BlockDataBasic) (bool, *chainhash.
 	}
 
 	// At capacity
-	for int(pq.capacity) <= pq.Len() {
+	if int(pq.capacity) <= pq.Len() {
```

Comment:
Was meant to be if.  The return statements guarantee there cannot be multiple iterations.